### PR TITLE
Add credit information to sidebar class data

### LINKED ISSUE #492: Semester Credits Display as NaN in the Plan
Screen Shot 2022-10-30 at 3 38 38 PM

Semester credits display as NaN in the plan when adding a course from the sidebar (or having preexisting courses from the sidebar). Currently reproducable on main.

### PR DESCRIPTION
**Pull request by AlpacaFur**
# Description

Small fix to get class credits information from Search's API. Also made missing courses display their subject + classId so it's easier to debug missing classes later and provides some extra info to users.

Closes #492

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Manually tested by checking that the credits add up properly in the semesters now.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/api-client/index.ts [hunk 1]
```diff
@@ -163,7 +165,17 @@ class SearchAPIClient {
...
     if (coursesData.bulkClasses) {
-      return coursesData.bulkClasses;
+      return coursesData.bulkClasses.map((course) => {
+        if (course === null) return null;
+
+        return {
+          name: course.name,
+          classId: course.classId,
+          subject: course.subject,
+          numCreditsMin: course?.latestOccurrence?.minCredits,
```
Comment (AlpacaFur) [inline packages/api-client/index.ts:175]: I had to unwrap the data here bc minCredits is nested within latestOccurence (and we have a different name for it).

Comment (aryanshah701) [reply packages/api-client/index.ts:175]: awesome, we do similar logic in the fetchCourse function... could you abstract it out? We use fetchCourse in getting course coreqs and things I think(may be time to move that to bulk as well but that's for a later task)

**Review by aryanshah701 — APPROVED** [review]
🚢🚢🚢

File: packages/api-client/index.ts (hunk 1 above)
Comment (AlpacaFur) [reply packages/api-client/index.ts:175]: will do!

**Pull request merged**